### ISSUE TITLE
Switch to *.docksal.dev for local environments

### ISSUE THREAD

**Issue by lmakarov** [docksal/docksal#1215]
There are various complications and workaround we had to make to get the wildcard `*.docksal` domain resolution work for local environments on different systems. To remove this complexities and have a simpler and more reliable setup, we will be deprecating the use of `*.docksal` and switch to the publicly resolvable `*.docksal.dev` domain.

There are multiple adjustments necessary before we can make the switch. We'll first have this as an option, available as early as 1.14 release. Forcing the switch won't happen until the 2.0 release.

**Comment by lmakarov** [comment]
One important caveat, or rather a show stopper: Google enforces https with a valid cert on all `.dev` domains in Chrome. A self-signed cert won't work. With that in mind, we will have to pick a different TLD.

**Comment by achekulaev** [comment]
`docksal.host`
`docksal.site`

**Comment by lmakarov** [comment]
`docksal.site` it is!

**Comment by achekulaev** [comment]
Some cybersquatter jerk registered `docksal.site`. I took `docksal.host` so that would not get taken too.

**Comment by achekulaev** [comment]
I guess the optional part is done now?

**Comment by lmakarov** [comment]
Right, the optional part has been released in [1.13.3](https://github.com/docksal/docksal/releases/tag/1.13.3) as a workaround for issues with the recent Docker Desktop version on Windows.

It can be used either globally for all projects:

```
fin config set --global DOCKSAL_DNS_DOMAIN=docksal.site
```

or per individual project:

```
fin config set DOCKSAL_DNS_DOMAIN=docksal.site
```

Note: This option is enforced with Docker Desktop for Windows 2.2.0.0+

**Comment by gconry18** [comment]
Out of interest, why is .docksal.site not the default now with a fresh docksal install?

**Comment by achekulaev** [comment]
Because it ties you to the internet connection, i.e. your local sites will in the plane or such. Also because of legacy that needs to be removed. Also we were hoping some way comes up that would make `.docksal` domain usable again. It does not seem to move in that direction though.

**Comment by MrUpsidown** [comment]
.site being a valid domain extension, could this not cause issues?

**Comment by lmakarov** [comment]
@MrUpsidown 

> .site being a valid domain extension, could this not cause issues?

We are using `*.docksal.site`, not simply `*.site`.

docksal.site is a real registered domain. The root record redirects to docksal.io. 

`*.docksal.site` points to `192.168.64.100` (Docksal's canonical IP) - it's public wildcard DNS record pointing to a local IP.


```
$ nslookup anything.docksal.site 8.8.8.8
Server:		8.8.8.8
Address:	8.8.8.8#53

Non-authoritative answer:
Name:	anything.docksal.site
Address: 192.168.64.100
```

**Comment by obriat** [comment]
Two problems :

- host.docker.internal is (still) not available inside docksal containers.
- 192.168.64.100 doesn't seems to point to my docker instance (certainly due to greedy VPN)

For the record, I use the following trick to declare a host inside a container project that points to  docksal. It could be useful to communicate with another docksal project
```
# .docksal/docksal-local.env 
DOCKSAL_HOST_IP=' 192.168.64.100'
```
```
# .docksal/docksal.yml
...
  cli:
  ...
    extra_hosts:
      - "${VIRTUAL_HOST}:${DOCKSAL_HOST_IP}"
  ...
```

I put the following line inside a command, I execute it before starting project (and each time I switch my VPN) :
```
sed -i "s/DOCKSAL_HOST_IP='[^']*'/DOCKSAL_HOST_IP='$(hostname -I | awk '{print $1}')'/g" .docksal/docksal-local.env
```
